### PR TITLE
Prevent crash when running snapshot and schema at the same time

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Also moved swapped the order of the schema and snapshot buttons to more conform to the order you'd run each command.

#### Primary reviewers
@danielimprobable @joshuahuburn

### CODE REVIEW

File: SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbar.cpp [hunk 1]
```diff
@@ -215,8 +220,8 @@ void FSpatialGDKEditorToolbarModule::AddMenuExtension(FMenuBuilder& Builder)
 {
 	Builder.BeginSection("SpatialOS Unreal GDK", LOCTEXT("SpatialOS Unreal GDK", "SpatialOS Unreal GDK"));
 	{
-		Builder.AddMenuEntry(FSpatialGDKEditorToolbarCommands::Get().CreateSpatialGDKSnapshot);
 		Builder.AddMenuEntry(FSpatialGDKEditorToolbarCommands::Get().GenerateSchema);
```
Comment (danielimprobable) [inline SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbar.cpp]: Should we call this function `CreateSpatialGDKSchema` now to be consistent?

File: SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbarCommands.cpp [hunk 2]
```diff
@@ -6,8 +6,8 @@
 
 void FSpatialGDKEditorToolbarCommands::RegisterCommands()
 {
-	UI_COMMAND(CreateSpatialGDKSnapshot, "Snapshot", "Creates SpatialOS Unreal GDK snapshot.", EUserInterfaceActionType::Button, FInputGesture());
 	UI_COMMAND(GenerateSchema, "Schema", "Generates schema.", EUserInterfaceActionType::Button, FInputGesture());
+	UI_COMMAND(CreateSpatialGDKSnapshot, "Snapshot", "Creates SpatialOS Unreal GDK snapshot.", EUserInterfaceActionType::Button, FInputGesture());
```
Comment (danielimprobable) [inline SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbarCommands.cpp]: Why did we change the order?

Comment (m-samiec) [reply SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbarCommands.cpp]: This was superficial, but to match the order they appear in the toolbar.

File: SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbar.cpp [hunk 3]
```diff
@@ -153,23 +153,28 @@ bool FSpatialGDKEditorToolbarModule::HandleSettingsSaved()
 	return true;
 }
 
-bool FSpatialGDKEditorToolbarModule::CanExecuteSchemaGenerator()
+bool FSpatialGDKEditorToolbarModule::CanExecuteSchemaGenerator() const
 {
 	return !bSchemaGeneratorRunning;
```
Comment (joshuahuburn) [inline SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbar.cpp]: Shouldn't we also check if SnapshotGenerator is running if we're doing this?

Comment (m-samiec) [reply SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorToolbar.cpp]: Snapshot generator isn't async, so all we care about is the schema generator.

**Pull request merged**